File: 1912/Domain/Card.cs
namespace _1912.Domain
{

    public class Card
    {
        public string ID { get; }
        public string Name { get; }
        public string Description { get; }
        public string ImageFolder { get; }

        public enum CardType
        {
            Battalion,
            Cannon,
            Operation
        };

        public CardType Type { get; }

        public int KnockStat { get; }
        public int ReplacementStat { get; }
        public int MainStat { get; }

        public Card(CardType type, string id, string name, string description, int knockStat = 0, int replacementStat = 0, int mainStat = 0)
        {
            Type = type;
            ID = id;
            Name = name;
            Description = description;
            KnockStat = knockStat;
            ReplacementStat = replacementStat;
            MainStat = mainStat;
            ImageFolder = "../../../Data/Images/CardImages/" + ID + ".png"; //проверить, написать норм путь
        }

        public Card() { }

        public override string ToString()
        {
            return Name;
        }
    }
}

File: domain/card.py
from enum import Enum

class Card:

    def __init__(self, type_id, id, name, description, knockStat = 0, replacementStat = 0, mainStat = 0) -> None:
        self.type_id = type_id
        self.type = CardType(type_id)
        self.id = id
        self.name = name
        self.description = description
        self.knock_stat = knockStat
        self.replacement_stat = replacementStat
        self.main_stat = mainStat
        self.image_folder = "../../../Data/Images/CardImages/" + id + ".png" #проверить, написать норм путь

    def __str__(self) -> str:
        return self.id

    def describe(self):
        return 'Тип: {}\tНазвание: {}\nОписание: {}\mKnock: {}\tReplacement: {}\tMain: {}'.format(self.type, self.name, self.description, self.knock_stat, self.replacement_stat, self.main_stat)

class CardType(Enum):
    Battalion = 0
    Cannon = 1
    Operation = 2